pkg/logger/json.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2022 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+)
+
+const (
+	// DefaultTimeFormat is the time format to use for JSON output
+	DefaultTimeFormat = "2006-01-02T15:04:05.000000Z07:00"
+
+	// errInvalidTypeMsg message implies an arg cannot be serialized to json
+	errInvalidTypeMsg = "cannot serialize arg(s) to json"
+
+	// defaultCallerOffset is the stack frame offset in the call stack for the caller
+	defaultCallerOffset = 3
+)
+
+var (
+	// DefaultOutput is used as the default log output.
+	DefaultOutput io.Writer = os.Stderr
```

Comment:
I'd use global variables only when they're meant to be externally set at compile time... otherwise it might lead to unexpected changes...

Reply:
This is not being set anywhere I was thinking this could be useful if a user using this library wants to use a set of preinitialised defaults. Wdyt?

Reply:
made this variable private